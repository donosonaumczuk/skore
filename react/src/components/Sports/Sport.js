import React from 'react';
import Proptypes from 'prop-types';
import SportInfo from '../match/SportInfo';
import SportPlayers from './SportPalyers';

const getSportImageUrl = links => {
    let sportImageUrl;
    links.forEach(link => {
        if (link.rel === "image") {
            sportImageUrl = link.href;
        }
    });
    return sportImageUrl;
}

const Sport = ({ sport }) => {
    //TODO maybe make image bigger and name bigger, add more staff to sport
    const sportImageUrl = getSportImageUrl(sport.links);
    return (
        <div className="row p-2 mt-2 ml-5 mr-5 match-card rounded-border">
            <div className="col">
                <div className="row mb-4">
                    <SportInfo sportImageUrl={sportImageUrl} sport={sport.displayName} />
                    <SportPlayers playersQuantity={sport.playerQuantity} />
                </div>
            </div>
        </div>
    );
}

Sport.propTypes = {
    sport: Proptypes.object.isRequired//TODO add custom sport proptype
}

export default Sport;