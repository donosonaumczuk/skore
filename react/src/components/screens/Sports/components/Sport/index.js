import React from 'react';
import SportInfo from '../../../../match/SportInfo';
import SportPlayers from '../SportPlayers';
import SportPropTypes from '../../../../../proptypes/SportPropType';

const getSportImageUrl = links => {
    let sportImageUrl;
    links.forEach(link => {
        if (link.rel === "image") {
            sportImageUrl = link.href;
        }
    });
    return sportImageUrl;
}

const Sport = ({ sport }) => {
    //TODO maybe make image bigger and name bigger, add more staff to sport
    const sportImageUrl = getSportImageUrl(sport.links);
    return (
        <div className="row p-2 mt-2 ml-5 mr-5 match-card rounded-border">
            <div className="col">
                <div className="row mb-4">
                    <SportInfo sportImageUrl={sportImageUrl} sport={sport.displayName} />
                    <SportPlayers playersQuantity={sport.playerQuantity} />
                </div>
            </div>
        </div>
    );
}

Sport.propTypes = {
    sport: SportPropTypes.isRequired
}

export default Sport;