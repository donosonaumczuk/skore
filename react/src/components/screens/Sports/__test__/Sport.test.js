import React from 'react';
import renderer from 'react-test-renderer';
import Sport from '../components/Sport';

test('Sport snapshot test', () => {
    //set up
    const sport = {
        sportName: "SportName",
        displayName: "Display name",
        playerQuantity: 1,
        links: [
            {
                rel: "image",
                "href": "sportImageUrl"
            }
        ]
    }

    //execution
    const component = renderer.create(
      <Sport sport={sport} />,
    );
    let tree = component.toJSON();

    //postconditions
    expect(tree).toMatchSnapshot();
});