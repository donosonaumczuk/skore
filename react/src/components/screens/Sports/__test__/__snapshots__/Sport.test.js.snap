// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sport snapshot test 1`] = `
<div
  className="row p-4 mt-2 ml-5 mr-5 match-card rounded-border"
>
  <div
    className="col"
  >
    <div
      className="row"
    >
      <div
        className="col-2 col-sm-3"
      >
        <div
          className="container-fluid pt-2"
        >
          <div
            className="row"
          >
            <div
              className="col col-xl-4 mr-0"
            >
              <img
                alt="sport-pic"
                className="sport-img"
                src="sportImageUrl"
              />
            </div>
            <div
              className="col-6 col-xl d-none d-sm-block pl-0 my-auto"
            >
              <p
                className="sport-label"
              >
                Display name
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="offset-1 col my-auto"
      >
        <i
          className="name-label fas fa-users mr-2"
        />
        <label />
        <span>
           1
        </span>
      </div>
      <div
        className="offset-1 col my-auto"
      />
    </div>
  </div>
</div>
`;
